sled-agent/src/bootstrap/agent.rs

```diff
@@ -729,6 +746,163 @@ impl Agent {
         Ok(())
     }
 
+    /// Runs the rack setup service to completion
+    pub async fn rack_reset(&self) -> Result<(), BootstrapError> {
+        RssHandle::run_rss_reset(&self.parent_log, self.ip, None).await?;
+        Ok(())
+    }
```

Comment:
Just to make the callstack a little more obvious:

- `sled-agent/src/bootstrap/http_entrypoints.rs` has a `rack_reset` endpoint, which calls...
- `sled-agent/src/bootstrap/agent.rs` 's `rack_reset`. This is a pass-through function which calls calls...
- `sled-agent/src/bootstrap/rss_handle.rs`'s `RssHandle::run_rss_reset`, which abstracts away some of the "local vs remote" details of accessing the bootstrap agent. This calls...
- `sled-agent/src/rack_setup/service.rs`'s `new_reset_rack`, which spawns a tokio task that calls...
- `sled-agent/src/rack_setup/service.rs`'s `RackSetupService::reset`. This function does the following:
  - Query for all visible bootstrap addresses from the current sled
  - Send a `reset` request to all of them (including our own!). This goes through...
- `sled-agent/src/bootstrap/rss_handle.rs` - `BootstrapAgentHandle::reset_sleds`, which sends reset requests to all sleds (including ourselves) in the same way that all the initialization requests are normally sent out.